core/trino-main/src/main/java/io/trino/execution/CreateMaterializedViewTask.java

```diff
@@ -119,7 +127,14 @@ public ListenableFuture<Void> execute(
                 Optional.empty(),
                 properties);
 
-        accessControl.checkCanCreateMaterializedView(session.toSecurityContext(), name, properties);
+        Set<String> specifiedPropertyKeys = statement.getProperties().stream()
+                // property names are case-insensitive and normalized to lower case
+                .map(property -> property.getName().getValue().toLowerCase(ENGLISH))
+                .collect(toImmutableSet());
+        Map<String, Object> explicitlySetProperties = properties.keySet().stream()
+                .filter(specifiedPropertyKeys::contains)
```

Comment:
lowercase here when invoking `contains`

Reply:
It is already lower cased in `io.trino.metadata.PropertyUtil#evaluateProperties`

Reply:
Can this class depend on that? maybe it should not, so that the code is "more obviously correct".
i.e. consisten with lowercase 3 lines up

Reply:
Also `io.trino.spi.session.PropertyMetadata#PropertyMetadata` disallows to create a property with non lower cased property name.

Reply:
> Also `io.trino.spi.session.PropertyMetadata#PropertyMetadata` disallows to create a property with non lower cased property name.

today.

if you rely on that other class guarantees, add a `verify` here (`.peek(...)`)